mypy/messages.py

```diff
@@ -1975,10 +1975,12 @@ def [T <: int] f(self, x: int, y: T) -> None
             s += ' = ...'
 
     # If we got a "special arg" (i.e: self, cls, etc...), prepend it to the arg list
-    if isinstance(tp.definition, FuncDef) and tp.definition.name is not None:
+    if isinstance(tp.definition, FuncDef) and \
+       tp.definition.name is not None and \
+       tp.definition.arguments:
```

Comment:
```suggestion
    if (isinstance(tp.definition, FuncDef) and
        tp.definition.name is not None and
        tp.definition.arguments):
```

Minor nit: I prefer parenthesis rather than `\` for wrapping multiple lines of if statements.

Reply:
Hi @ethanhs ,

Thanks for the review. I've made that change. In fact I've also simplified the next check: since we check for `tp.definition.arguments` on line 1980, there is no need to check for `len(definition_args) > 0` a couple of lines down.